enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+- We want to make sure that the default behavior of this feature is to allow
+  full use of the CPU set.
+- We want to maintain the ability to configure the partition size after
+  installation, we do not support turning off the partition feature but we do
+  support changing the CPU partition size post day 0. The ability for turning on
+  this feature will be part of the installation phase only.
+
+### Non-Goals
+
+This enhancement expands on the existing [Management Workload
+Partitioning](management-workload-partitioning.md) and as such shares similar
+but slightly different non-goals.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+- This enhancement is focused on CPU resources. Other compressible resource
+  types may need to be managed in the future, and those are likely to need
+  different approaches.
+- This enhancement does not address non-compressible resource requests, such as
+  for memory.
+- This enhancement does not address ways to disable operators or operands
+  entirely.
+- This enhancement does not address reducing actual utilization, beyond
+  providing a way to have a predictable upper-bounds. There is no expectation
+  that a cluster configured to use a small number of cores for management
+  services would offer exactly the same performance as the default. It must be
+  stable and continue to operate reliably, but may respond more slowly.
+- **This enhancement does not address mixing nodes with pinning and without,
+  this feature will be enabled cluster wide and encapsulate both master and
+  worker pools. If it's not desired then the default behavior will still be
+  turned on but the management workloads will run on the whole CPU set for that
+  desired pool.**
+- **This enhancement assumes that the configuration of a management CPU pool is
+  done as part of installing the cluster. It can be changed after the fact but
+  we will need to stipulate that, that is currently not supported. The intent
+  here is for this to be supported as a day 0 feature, only.**
+- **This enhancement does not address mixing nodes of different CPU sizes in
+  Machine Configuration Pools. Pools must contain machines of the same CPU set size,
+  currently the Performance Profile defines and targets specific machine pools
+  to apply the configuration to, as such they must have the same CPU set size.**
+
+## Proposal
+
+We will need to maintain a global identifier that is set during installation and
+can not be easily removed after the fact. This approach will help remove
+exposing this feature via an API and limit the chances that a misconfiguration
+can cause un-recoverable scenarios for our customers. At install time we will
+also apply an initial machine config for workload partitioning that sets a
+default CPUSet for the whole CPUSet. Effectively this will behave as if workload
```

Comment:
We currently don't have a way to represent that since that value as is gets handed over to the CPUSet Parser for validation. That parser follows the rules defined https://man7.org/linux/man-pages/man7/cpuset.7.html#FORMATS so we would need to create a special identifier like `*` or `-1` to mean all cores and then behind the scenes at the lower level strip the CPUSet value so system defaults kick in.

Reply:
When PerformanceProfile is used to cpu partition a MCP the MCP hardware MUST be homogenous. This has always been the case.

Reply:
That's a great point, updated to make this clear with the new direction that we will not be changing the Performance Profile API.